plugin/trino-phoenix/src/main/java/io/trino/plugin/phoenix/PhoenixSplitManager.java

```diff
@@ -154,10 +155,11 @@ public ConnectorSplitSource getSplits(
                  * Each split is handled by a single ConcatResultIterator
                  * (See PhoenixClient.getResultSet(...))
                  */
-                for (List<Scan> splitScans : Lists.partition(scans, MAX_SCANS_PER_SPLIT)) {
+                for (List<Scan> splitScans : Lists.partition(scans, maxScansPerSplit)) {
                     inputSplits.add(new PhoenixInputSplit(splitScans, regionSize, regionLocation));
                 }
             }
+            Collections.shuffle(inputSplits);
```

Comment:
Why is this needed, not obvious to me.

Reply:
We now may get multiple Splits for the same region, and it seems Trino runs them in order per worker. (Right?)
We want to avoid a worker running multiple splits against the same region concurrently while splits for other regions are waiting in the queue.

Reply:
Add this as a comment. cc: @sopel39 to confirm split execution ordering, I'm not very familiar with the scheduler off the top of my head.